src/shared/helpers/SubscriptionHelper.ts

```diff
@@ -11,40 +11,18 @@ import Log from '../libraries/Log';
 import { ContextSWInterface } from '../models/ContextSW';
 import SdkEnvironment from '../managers/SdkEnvironment';
 import { EnvironmentInfo } from '../../page/models/EnvironmentInfo';
-import { SessionOrigin } from '../models/Session';
 import { Browser } from '../models/Browser';
-import { PushDeviceRecord } from '../models/PushDeviceRecord';
 import { PermissionUtils } from '../utils/PermissionUtils';
-import MainHelper from './MainHelper';
 
 export default class SubscriptionHelper {
   public static async registerForPush(): Promise<Subscription | null> {
-    const isPushEnabled = await OneSignal.context.subscriptionManager.isPushNotificationsEnabled();
-    return await SubscriptionHelper.internalRegisterForPush(isPushEnabled);
+    return await SubscriptionHelper.internalRegisterForPush();
   }
 
-  public static async internalRegisterForPush(isPushEnabled: boolean): Promise<Subscription | null> {
+  public static async internalRegisterForPush(): Promise<Subscription | null> {
     const context: ContextSWInterface = OneSignal.context;
     let subscription: Subscription | null = null;
 
-    /*
-      Within the same page navigation (the same session), do not register for
-      push if the user is already subscribed, otherwise the user will have its
-      session count incremented on each page refresh. However, if the user is
```

Comment:
Did you confirm the session count is not incrementing even after the refresh? What is the impact on other browsers? From first glance it doesn't appear this code was Safari-specific.

Reply:
I updated the "Tests" section in this PR to provide more details on this.